src/main/java/seedu/address/model/person/DateOfBirth.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.DateUtil.isAfterToday;
+import static seedu.address.commons.util.DateUtil.isValidDate;
+
+/**
+ * Represents a patient's date of birth in MediBase3.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDateOfBirth(String)}
+ */
+public class DateOfBirth {
+    public static final String MESSAGE_CONSTRAINTS_FUTURE_DATE = "Date of birth should not be after today's date";
+    public static final String MESSAGE_CONSTRAINTS_WRONG_FORMAT = "Date of birth should be in the format of yyyy-MM-dd";
+    public final String value;
+
+    /**
+     * Constructs a {@code DateOfBirth}.
+     *
+     * @param dob A valid date of birth.
+     */
+    public DateOfBirth(String dob) {
+        requireNonNull(dob);
+        checkArgument(isValidDate(dob), MESSAGE_CONSTRAINTS_WRONG_FORMAT);
+        checkArgument(isValidDateOfBirth(dob), MESSAGE_CONSTRAINTS_FUTURE_DATE);
+        this.value = dob;
+    }
```

Comment:
Proper use of the SLAP principle. Good job on abstracting the checks for valid date and DateOfBirth.

Reply:
I second this 😀

Reply:
I third this